### DISCUSSION TITLE
Dexcom G6 Transmitter Calibration Issues

### DISCUSSION THREAD

**Discussion by snoopstaa** [NightscoutFoundation/xDrip#3690] [Bug reports] [answered]
Hi,

I know the issues I'm experiencing with my calibrations not working are not an issue with the excellent xDrip+ and more likely the transmitter as I can see the calibration accepted by transmitter events in the logs and I've checked and I'm running in Native mode.

I do have a quick question though as Dexcom have stated that the transmitter does not accept calibrations, so my question is does the Dexcom G6 app run in Native mode like xDrip+ or might it use a different mode and it is the app that stores and manages the calibrations entered?

**Comment by Navid200** [comment, marked as answer]
What do you mean by Dexcom having stated that the transmitter does not accept calibrations?

Calibration, whether it comes from the receiver or Dexcom app or xDrip, is submitted to the G6 transmitter.  The transmitter deals with the calibration.

In the early days, G6 transmitters transmitted raw values the same as G5.  That was when xDrip could use a G6 in non-native mode and handle calibrations itself.  But, G6 introduced the concept of factory calibration, which addressed the impact of the immune system on the behavior of the sensor.  This capability never existed in G5.  Using an old G6 in non-native mode dismissed factory calibration, a major evolution over G5.  Therefore, it was not a very good idea to use a G6 in non-native mode.
Later on, G6 production was modified and the new transmitters did not transmit raw values any longer.  Therefore, it is not possible to use a new G6 in non-native mode.  

Please explain if I have misunderstood your question.

**Comment by snoopstaa** [comment]
Hi Navid200,

Yes you have answered my question perfectly.

I have a G6 transmitter that was working fine for a month or so, normally it runs a little high when a new sensor is inserted and I have to calibrate once or twice the next day and it's perfect from then.

However, after starting a new sensor, it is about 5 - 7 mmols too high, I've tried to calibrate it (within 20%) and the error log says calibration accepted but it is having zero effect on the subsequent readings. Today is my 4th day of the sensor and every calibration has had zero effect. Every 5 min the reading is ~5mmol above what is recorded by my BGM (normally once settled they are largely the same). I even had a minor hypo where my G6 showed 5mmol (alarm went off) and my BGM showed 1.2mmol.

I phoned Dexcom in Aus (AMSL) and was told it was definitely a sensor issue (ok, being ~5mmol to high maybe) but the issue is that the transmitter isn't handling the calibration event (usually the readings align / move / gprah over the next few readings). Suddenly though  there is zero effect of calibrations to correct high values (even when my blood readings are stable and when I even tried calibrating down by 2mmol to be within 20%). I would be happy even with a 3mmol correction tbh, but 5mmol is too high for me.

When I said the issue was the transmitter not reacting to the calibrations they said categorically the sensor handled this. I argued and said it can't be as the sensor doesn't have a chip in it, they then said the transmitter does not do anything with the calibrations and that it is all done by the app and I needed to restart my app.

Having become acquainted with the event logs on xDrip+ and was aware of the calibration being sent to the transmitter event I highly doubted that so thought I'd seek understanding from you guys, but clearly they are talking nonsense.

I'm just trying to get a better understanding for the likely fight I'm going to have with the technician (who still hasn't called back) for a transmitter replacement.

Btw, I'm aware of over calibration, calibration errors, calibrating within 20% and calibration confusion, my graph looks perfect and is graphing every 5 mins, the sole issue I'm having is every reading is ~5mmols too high and the calibrations I'm providing, being accepted by the transmitter are having zero effect.

Sorry for the long response, thought you might appreciate the detail.

**Reply by Navid200** [reply]
Do you mean you see a difference of 5 mmol/L?!  
So, for example, your meter measures 5.5mmol/L while the CGM measures 10.5mmom/L?

When you call Dexcom, you don't always get someone who is literally a technician.  I don't believe so.

Are you using a G6?  Or a Dexcom One?

Is it possible that you are not using an xDrip from this repository?

Calibrations will have lower and lower weights the more calibrations you perform.  This is understandable because the device cannot forget your previous calibrations.  There is more than one parameter to adjust.  So, there is no way to adjust both with a single correction.  That's why the CGM must maintain all or at least, some (recent) calibrations and add every new entry into the set.  It will be the entire set that defines the calibration constants.  

Is this something that you have experienced with only one sensor?  Or have you experienced it with more than one sensor?

Is it possible that your blood glucose meter is measuring somewhat too low?

**Reply by snoopstaa** [reply]
Hi,

Yes, my G6 states I am 17mmol and my BGM states I am 9.8mmol (just took these readings).

Glucose meter is the same I always use, I even changed to another box of strips to check the strips.
My hypo symptoms at 1.2mmol on my BGM indicate it is correct (G6 said 5mmol, which is not low enough for me to have hypo symptoms).

I'm also not sick and tbh 17mmol is way too high a reading for me at the moment, 9.8mmol feels correct as I ate dinner 1 hour ago. I have no reason to doubt my BGM, I noticed the G6 was wrong pretty quickly and immediately started testing.

I'm fairly lean and normally the G6 sensor usually initially presents a little high for me, that's why I have to calibrate a couple of times every new sensor, but it's always been pretty accurate after that. But, without the calibration working on the transmitter though, it's just stayed high for the whole of this sensor (4 days so far).

FYI, I found a paper that specifically states that the G6 transmitter handles the algo [https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9445335/](url)

I am using 645e8b4-2024.04.19 version from this repo.

I have probably performed too many calibrations now, but from the start of this new sensor no calibrations had any effect on subsequent readings and it's only got worse.

**Reply by Navid200** [reply]
The unit of glucose is mmol/L.  Some like to use lowercase for L, mmol/l.  But, mmol is not a valid glucose unit and could be very confusing.

You did not answer my question whether this was something you had experienced with your current transmitter with different sensors or only with the current sensor.
If you insert a sensor, but do not start the session right away, the readings can be higher than they should be when you eventually start.  
Regardless, you can calibrate if you avoid entering calibrations that are too different.  If you enter a calibration that is too far off (more than 20%), the transmitter could ignore it.

Yes calibrations are handled by the transmitter. 
But, the 4-digit calibration code is printed on the sensor.  And when you start a sensor, you must enter that code correctly.  if you don't, you could end up with incorrect readings.  Sensors are binned in manufacturing to different calibration codes.  This association of the sensor with a particular calibration code may be the cause of the confusion when you spoke with the technician.

**Reply by snoopstaa** [reply]
Yes, I mean mmol/L just thought most diabetics would recognise mmol as an abbreviation.

My last sensor didn't seem to accept a couple of calibrations near the end of its life, but didn't really bother as it was on day 8.

I know about the 20% rule, I tried to calibrate it down using a larger value than my BGM reading so it was in bounds, but the readings still didn't budge.

All codes are entered correctly, I already checked on system status.

I think she was just call staff operator, but she was adamant that the calibration entered into the 'app' was sent to the sensor (despite there being no chip), and then stated the calibration was just held by the app and not transmitted at all. She was very adamant there was no issue with the transmitter, up to the point of contradicting herself. It took a while for me to convince her to get a technician to call me back which still hasn't happened.
